.github/workflows/test-framework.yml

```diff
@@ -18,25 +18,41 @@ on:
 
 jobs:
   test:
-    name: run tests
+    name: ${{ matrix.os }}, node ${{ matrix.node-version }}, ${{ matrix.browser }}
     runs-on: ${{ matrix.os }}
+    if: "!contains(github.event.head_commit.message, 'skip ci')"
 
     strategy:
       matrix:
-        os: [ubuntu-latest, windows-latest]
+        os: [ubuntu-latest, windows-latest, macos-latest]
         node-version: [10.x, 12.x, 14.x]
+        browser: [no-browser, chromium, firefox, webkit]
 
     steps:
       - name: checkout
         uses: actions/checkout@v2
 
-
       - name: use node.js ${{ matrix.node-version }}
         uses: actions/setup-node@v1
         with:
           node-version: ${{ matrix.node-version }}
 
-      - name: install and test
+      - name: configure MacOS
+        if: "matrix.os == 'macos-latest'"
+        run: ulimit -f unlimited
+
+      - name: bootstap framework
         run:  |
               npm  ci
-              npm  test -- --browsers=chromium,firefox --terse
+              ./bootstrap-compiler --target=build
+
+      - name: install and run browser tests with ${{ matrix.browser }}
+        if: "matrix.browser != 'no-browser' && !(matrix.os == 'windows-latest' && matrix.browser == 'webkit')"
+        run:  |
+              npm test -- --browsers=${{ matrix.browser }} --terse --set-env qx.test.delay.scale=5
+
+      - name: install and run node tests
+        if: "matrix.browser == 'no-browser'"
+        run:  |
+              cd test/tool
```

Comment:
npm test will only work in the root.

Reply:
grr, this is a leftover from the previous attempt. thanks.